### PR TITLE
User Menu 2.0

### PR DESCRIPTION
**Pull request by bromiesTM**

### CODE REVIEW

File: IONOS/src/components/Icon.svelte [hunk 1]
```diff
@@ -84,13 +88,64 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 		<path
 			d="M1.5 0H14.5C15.3125 0 16 0.6875 16 1.5C16 2 15.75 2.4375 15.375 2.71875L8.59375 7.8125C8.21875 8.09375 7.75 8.09375 7.375 7.8125L0.59375 2.71875C0.21875 2.4375 0 2 0 1.5C0 0.6875 0.65625 0 1.5 0ZM0 3.5L6.78125 8.625C7.5 9.15625 8.46875 9.15625 9.1875 8.625L16 3.5V10C16 11.125 15.0938 12 14 12H2C0.875 12 0 11.125 0 10V3.5Z" />
 	</svg>
+{:else if user}
+	<svg
+		id="user"
+		width={size * 24}
+		height={size * 26}
```
Comment (bromiesTM) [inline IONOS/src/components/Icon.svelte:95]: todo: seems like the size is incorrect
According to the mockup, the width and height should have an equal ratio

File: core/templates/layout.user.php [hunk 2]
```diff
@@ -64,7 +65,49 @@
...
+					<a href="<?php p($link) ?>"
+						target="_blank" slot="webmail"
+						title="<?php p($l->t('IONOS WEBMAIL')) ?>" data-qa="IONOS-WEBMAIL-TARGET">
+						<ionos-icons webmail size="1.6" />
+					</a>
+				<?php } ?>
+				<div class="usermenu" slot="usermenu" data-qa="IONOS-USER-MENU-TARGET">
+					<ionos-user-menu>
+						<b slot="user-name">
+							<?php p($_['user_displayname']); ?>
```
Comment (bromiesTM) [inline core/templates/layout.user.php:82]: todo: unstyled username is the first thing visible during page load 

same with `OCP\IUser::getDisplayName`

File: core/css/server.css [hunk 3]
```diff
@@ -19,13 +19,24 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 header#ionos-global-nav {
 	--color-main-background: #fff;
+	--color-blue-ionos-b1: #dbedf8;
+	--color-blue-ionos-b2: #95caeb;
+	--color-blue-ionos-b4: #1474c4;
+	--color-cool-grey-ionos-c3: #bcc8d4;
+	--typo-color-mild: #2e4360;
+	--color-lightgray: #d7d7d7;
+	--text-color: #001b41;
```
Comment (bromiesTM) [inline core/css/server.css:28]: suggestion: we should think about removing brand colours from the server.css
sth. like this https://github.com/IONOS-Productivity/nc-ionos-theme/commit/18317b7118cf3552a90d0f7f9fd4eefaa409d222

**Review by fracado — CHANGES_REQUESTED** [review]
Review done.

File: IONOS/Containerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# SPDX-FileCopyrightText: Kai Henseler <kai.henseler@strato.de>
+# SPDX-FileCopyrightText: 2024 STRATO AG
```
Comment (fracado) [inline IONOS/Containerfile:1]: I'm not sure if this complies with the [license header guidelines](https://docs.nextcloud.com/server/latest/developer_manual/getting_started/codingguidelines.html#license-headers)... :thinking: 
@thlehmann-ionos What do you think?

Comment (thlehmann-ionos) [reply IONOS/Containerfile:1]: They even have an example with a company name.

Comment (fracado) [reply IONOS/Containerfile:1]: I know... I wasn't sure if in that case we also need something like the "and contributors" part or not. Or if we have to add our names to that mentioned AUTHORS.md file or not?

Comment (thlehmann-ionos) [reply IONOS/Containerfile:1]: > ... if the author is being paid by their employer to create a work, the employer is often the copyright holder.

Source: https://reuse.software/faq/#copyright-holder-author

Referring to the [specification][spec] I read that as:

```
SPDX-FileCopyrightText: 2024 STRATO AG
SPDX-License-Identifier: AGPL-3.0-or-later
SPDX-FileContributor: Kai Henseler <kai.henseler@strato.de>
```

[spec]: https://spdx.github.io/spdx-spec/v2.3/file-tags/

Comment (fracado) [reply IONOS/Containerfile:1]: Ah... yeah, let's do it like that, if you agree @bromiesTM. Thanks @thlehmann-ionos! :slightly_smiling_face:

Comment (bromiesTM) [reply IONOS/Containerfile:1]: Well then you should add yourself too as I copied your work 🤔

Comment (bromiesTM) [reply IONOS/Containerfile:1]: [Done](https://github.com/IONOS-Productivity/nc-ionos-theme/compare/ea0d20ab61dfba883d80c8ebc6447234b9f76ef3..6fee4f0e69406efa43bf715d5ab72a7c9acf94cb)

**Review by fracado — APPROVED** [review]

**Review by fracado — APPROVED** [review]
review OK

**Pull request merged**